### PR TITLE
PIN-2767: remove agreements filter from getProducerEServices

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

**Review by TonioGela — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -2768,16 +2768,6 @@ paths:
           description: Query to filter EServices by name
           schema:
             type: string
-        - in: query
-          name: consumersIds
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: I still don't remember why this parameter has been added, I don't find it in the initial requirement, but are we sure it is not used by the frontend?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala [hunk 2]
```diff
@@ -293,34 +289,6 @@ final case class EServicesApiServiceImpl(
     }
   }
 
-  private def getProducerEServicesIds(producerId: UUID, consumersUUIDs: List[UUID])(implicit
-    contexts: Seq[(String, String)]
-  ): Future[List[UUID]] = getAllAgreements(producerId, consumersUUIDs)
-    .map(_.map(_.eserviceId).distinct)
-
-  private def getAllAgreements(producerId: UUID, consumerUUIDs: List[UUID])(implicit
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]: ⚠️ this is now used by a recently merged PR

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala (hunk 2 above)
Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala]: Ok, I'm going to rebase.

File: src/main/resources/interface-specification.yml (hunk 1 above)
Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml]: `consumerIds` is actually used.
I need to change the PR.

**Review by galales — APPROVED** [review]

**Pull request merged**